components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/FirefoxAccount.kt

```diff
@@ -99,24 +99,30 @@ class FirefoxAccount(handle: FxaHandle, persistCallback: PersistCallback?) : Aut
      * This performs network requests, and should not be used on the main thread.
      *
      * @param scopes List of OAuth scopes for which the client wants access
+     * @param entrypoint to be used for data collection
      * @return String that resolves to the flow URL when complete
      */
-    fun beginOAuthFlow(scopes: Array<String>): String {
+    fun beginOAuthFlow(scopes: Array<String>, entrypoint: String): String {
```

Comment:
As I mentioned in Slack I am conflicted about making `entrypoint` optional VS enforcing consumers to pass something in here. Probably leaning towards allowing to skip it and having our logic pass in `none` or `notset` as part of the slow.

Reply:
However our other methods already force users to set `entrypoint`, e.g in fxa_get_manage_account_url

Reply:
In addition we need to support utm parameters here and flow id parameters here if we want parity with the desktop implementation.

Params such as 

```
    // {String} flow_id - Flow Id, such as '5445b28b8b7ba6cf71e345f8fff4bc59b2a514f78f3e2cc99b696449427fd445'
      // {Number} flow_begin_time - Flow begin timestamp, such as 1590780440325
      // {String} device_id - Device Id, such as '7e450f3337d3479b8582ea1c9bb5ba6c'
```

Reply:
So full support for all parameters will look like, (i hope I got them all):

```
beginOAuthFlow(scopes, [entrypoint], [flow_id], [flow_begin_time], [device_id], [utm_source], [utm_content], [utm_medium], [utm_term], [utm_campaign]
```

Reply:
After chatting with @rfk, let's split out the optional parameters in a different issue and just fix the `entrypoint` here. I filed that issue for optional parameters here: https://github.com/mozilla/application-services/issues/3278

Reply:
And we are going ahead with making `entrypoint` a REQUIRED argument for consumers.